### PR TITLE
fix: store remote agent and protocol version during identify

### PR DESCRIPTION
**Pull request by achingbrain**
These values were not being stored during identify.

### CODE REVIEW

**Comment by achingbrain** [comment]
I did have a couple of questions about this - why is `message.agentVersion` stored PascalCase style as `"AgentVersion"`, I would have thought `"agentVersion"` would be more idiomatic.

Why are they stored as `Uint8Array`s when they are deserialized from the incoming message as protobuf strings?

**Comment by vasco-santos** [comment]
> I did have a couple of questions about this - why is message.agentVersion stored PascalCase style as "AgentVersion", I would have thought "agentVersion" would be more idiomatic?

I tracked it down to https://github.com/libp2p/js-libp2p/pull/724 . I am not sure, probably to be similar to Go? We did try to follow go whenever possible for storing stuff in the stores.

> Why are they stored as Uint8Arrays when they are deserialized from the incoming protobuf message as strings?

Because we persist the metadataBook content as Uint8Array, as a Datastore Value. In this case, we considered that anything could be stored as metadata, including a serialized protobuf

File: test/identify/index.spec.js [hunk 1]
```diff
@@ -430,6 +430,39 @@ describe('Identify', () => {
...
+      // Wait for peer store to be updated
+      // Dialer._createDialTarget (add), Identify (consume)
+      await pWaitFor(() => peerStoreSpyConsumeRecord.callCount === 1 && peerStoreSpyAdd.callCount === 1)
+      expect(libp2p.identifyService.identify.callCount).to.equal(1)
+
+      // The connection should have no open streams
+      await pWaitFor(() => connection.streams.length === 0)
+      await connection.close()
+
+      const remotePeer = PeerId.createFromB58String('12D3KooWHFKTMzwerBtsVmtz4ZZEQy2heafxzWw6wNn5PPYkBxJ5')
```
Comment (vasco-santos) [inline test/identify/index.spec.js:457]: Can we do not have the peerId hardcoded?

Comment (achingbrain) [reply test/identify/index.spec.js:457]: It's taken from https://github.com/libp2p/js-libp2p/blob/master/test/fixtures/browser.js#L6 where it's also hardcoded?

Comment (achingbrain) [reply test/identify/index.spec.js:457]: I thought I would have been able to get it from [remoteLibp2p](https://github.com/libp2p/js-libp2p/blob/master/test/identify/index.spec.js#L392) but that var does't appear to be set anywhere in the tests?

Comment (vasco-santos) [reply test/identify/index.spec.js:457]: I was meaning, probably more bullet proof to extract the id from the multiaddr

Comment (achingbrain) [reply test/identify/index.spec.js]: Aha, have updated it.

**Review by vasco-santos — APPROVED** [review]
LGTM

**Pull request merged**